sanic/blueprints.py

```diff
@@ -284,6 +287,11 @@ def register_middleware(_middleware):
             middleware = args[0]
             args = []
             return register_middleware(middleware)
+        elif len(args) == 2 and len(kwargs) == 0 and callable(args[0]):
```

Comment:
this check condition is a little bit error-prone...

Reply:
@yunstanford No need to check for `kwargs` ? Or you don't like the idea of `len(args)` check? A bit more context would help me make the necessary changes.

Reply:
@yunstanford Would you mind adding a bit more context to this item so that I can address it the right way. Thanks again

Reply:
Yeah, none of these conditions is specific to register blueprint group. I'm a little worry about that this'd be hard to maintain in the future when adding more things.

Reply:
@yunstanford Understandable. Logically speaking, we should completely get rid of this entire check section. It should be a generic implementation where we pass everything we get to the method and let method decide how it wants to handle. 

Let me see what I can do. I don't want to change the existing one too much

Reply:
@yunstanford I tweaked the way the blueprint group middleware was being identified to make it more definitive instead of the length check as you suggested. Please let me know if this looks okay.